app/react-native/src/preview/components/OnDeviceUI/index.js

```diff
@@ -1,24 +1,116 @@
-  const { stories, events, url } = props;
+export default class OnDeviceUI extends Component {
+  constructor(props, ...args) {
+    super(props, ...args);
 
-  return (
-    <View style={style.main}>
-      <View style={style.leftPanel}>
-        <StoryListView stories={stories} events={events} />
-      </View>
-      <View style={style.rightPanel}>
-        <View style={style.preview}>
-          <StoryView url={url} events={events} />
+    this.state = {
+      menuAnimation: new Animated.Value(0),
+      isMenuOpen: false,
+      selectedKind: null,
+      selectedStory: null,
+    };
+
+    this.storyChangedHandler = this.handleStoryChanged.bind(this);
+    this.handleToggleMenu = this.handleToggleMenu.bind(this);
+
+    this.props.events.on('story', this.storyChangedHandler);
+  }
+
+  componentWillUnmount() {
+    this.props.events.removeListener('story', this.storyChangedHandler);
+  }
+
+  handleStoryChanged(storyFn, selection) {
+    const { kind, story } = selection;
+    this.setState({
+      selectedKind: kind,
+      selectedStory: story,
+    });
+  }
+
+  handleToggleMenu() {
+    const isMenuOpen = !this.state.isMenuOpen;
+
+    Animated.timing(this.state.menuAnimation, {
+      toValue: isMenuOpen ? 1 : 0,
+      duration: 150,
+      easing: Easing.linear,
+    }).start();
+
+    this.setState({
+      isMenuOpen,
+    });
+  }
+
+  render() {
+    const { stories, events, url } = this.props;
+    const { isMenuOpen, menuAnimation, selectedKind, selectedStory } = this.state;
+
+    const overlayStyles = [style.overlayContainer, { opacity: menuAnimation }];
+
+    const menuStyles = [
+      style.menuContainer,
+      {
+        transform: [
+          {
+            translateX: menuAnimation.interpolate({
+              inputRange: [0, 1],
+              outputRange: [0, 250],
+            }),
+          },
+        ],
+      },
+    ];
+
+    /* eslint-disable global-require */
+    const openMenuImage = require('./menu_open.png');
+    const closeMenuImage = require('./menu_close.png');
+    /* eslint-enable global-require */
+
+    return (
+      <View style={style.main}>
+        <View style={style.previewContainer}>
+          <View style={style.headerContainer}>
+            <TouchableWithoutFeedback onPress={this.handleToggleMenu}>
+              <View>
+                <Image source={openMenuImage} style={style.icon} />
+              </View>
+            </TouchableWithoutFeedback>
+            <Text style={style.headerText} numberOfLines={1}>
+              {selectedKind} / {selectedStory}
+            </Text>
+          </View>
+          <View style={style.previewWrapper}>
+            <View style={style.preview}>
+              <StoryView url={url} events={events} />
+            </View>
+          </View>
         </View>
+        {isMenuOpen &&
+          <TouchableWithoutFeedback onPress={this.handleToggleMenu}>
+            <Animated.View style={overlayStyles} />
+          </TouchableWithoutFeedback>}
```

Comment:
Prettier is not a fan of that. I typically would have it on the next line but prettier complained at me. I'll look at options because I agree that this isn't as readable as it could be.